samples/go/downloader/resource_cache.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/attic-labs/noms/go/datas"
+	"github.com/attic-labs/noms/go/spec"
+	"github.com/attic-labs/noms/go/types"
+)
+
+// ResourceCache is a Map<String, Ref<Blob>>
+type ResourceCache struct {
+	cache types.Map
+	orig  types.Map
+	mutex sync.Mutex
+}
+
+func GetResourceCache(db datas.Database, dsname string) (*ResourceCache, error) {
+	m, ok := db.GetDataset(dsname).MaybeHeadValue()
+	if ok {
+		refOfBlobType := types.MakeRefType(types.BlobType)
```

Comment:
I'm not sure I understand your point here. I don't want to unmarshal the cache into a go map because it could be very large. Can you explain this in more detail?